kornia/augmentation/_2d/intensity/salt_pepper_noise.py

```diff
@@ -0,0 +1,146 @@
+
+    Attributes:
+        _param_generator (PlainUniformGenerator): Parameter generator for random parameter generation.
+
+    Raises:
+        ValueError: If `salt_vs_pepper` or `amount` are not in the valid range.
+
+    Note:
+        The `amount` and `salt_vs_pepper` parameters control the intensity of the salt-and-pepper noise.
+        `amount` determines the range of noise values to be added, and `salt_vs_pepper` determines the
+        range of the salt vs. pepper ratio.
+
+    Example:
+        >>> transform = RandomSaltAndPepperNoise(amount=(0.02, 0.1), salt_vs_pepper=(0.3, 0.7), p=1.0)
+        >>> noisy_image = transform(input_image)
+    """
+
+    def __init__(
+        self,
+        amount: Optional[Union[float, Tuple[float, float]]] = (0.01, 0.06),
+        salt_vs_pepper: Optional[Union[float, Tuple[float, float]]] = (0.4, 0.6),
+        p: float = 0.5,
+        same_on_batch: bool = False,
+        keepdim: bool = False,
+    ) -> None:
+        super().__init__(p=p, same_on_batch=same_on_batch, p_batch=1.0, keepdim=keepdim)
+
+        if isinstance(salt_vs_pepper, (tuple, float)):
+            if isinstance(salt_vs_pepper, float):
+                salt_vs_pepper = (salt_vs_pepper, salt_vs_pepper)
+            elif len(salt_vs_pepper) == 1:
+                salt_vs_pepper = (salt_vs_pepper[0], salt_vs_pepper[0])
+            elif len(salt_vs_pepper) > 2 or len(salt_vs_pepper) <= 0:
+                raise ValueError(
+                    "The length of salt_vs_pepper must be greater than 0 \
+                        and less than or equal to 2, and it should be a tuple."
+                )
+        else:
+            raise ValueError("salt_vs_pepper must be a tuple or a float")
+        KORNIA_CHECK(
+            all(0 <= el <= 1 for el in salt_vs_pepper),
+            "Salt_vs_pepper values must be between 0 and 1. \
+                        Recommended value 0.5.",
+        )
+
+        if isinstance(amount, (tuple, float)):
+            if isinstance(amount, float):
+                amount = (amount, amount)
+            elif len(amount) == 1:
+                amount = (amount[0], amount[0])
+            elif len(amount) > 2 or len(amount) <= 0:
+                raise ValueError(
+                    "The length of amount must be greater than 0 \
+                        and less than or equal to 2, and it should be a tuple."
+                )
+        else:
+            raise ValueError("amount must be a tuple or a float")
+        KORNIA_CHECK(
+            all(0 <= el <= 1 for el in amount),
+            "amount of noise values must be between 0 and 1. \
+                        Recommended values less than 0.2.",
+        )
+
+        self._param_generator = rg.PlainUniformGenerator(
+            (amount, "amount", None, None), (salt_vs_pepper, "salt_vs_pepper", None, None)
+        )
+
+    def _generate_params(self, input: Tensor) -> Tuple[Tensor, Tensor]:
```

Comment:
please refer to https://github.com/kornia/kornia/blob/main/kornia/augmentation/random_generator/_2d/random_rain.py and https://github.com/kornia/kornia/blob/e427cc7dd3a174704f72e9a923fb2996da61429a/kornia/augmentation/_2d/intensity/random_rain.py#L47

to implement the random generators.

Reply:
+1